ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcHttpService.java

```diff
@@ -279,6 +310,32 @@ private Router buildRouter() {
     return router;
   }
 
+  private void createSpan(final RoutingContext routingContext) {
```

Comment:
Is OpenTelemetry "always on"?  Should there be a way for the end user to "opt out" like we do for other metrics collection?

Reply:
Asking for the best way on this.

Reply:
If it was just the one create span call it would be easy, but it looks like every request has to possibly add a span, so plumbing the metrics configs through seems a bit much... unless performance suffers.  We have a benchmark where rpcs is the bottleneck, so if it's a problem I'd expect those benchmarks to fail.  If you are willing to risk a call for a revert if performance takes a hit it can stay as-is.

Reply:
So I'm asking, and so far if we don't use a metrics backend, the spans are created, but in the words of OpenTelemetry devs, they then "fall to the floor" as there is no export enabled.
If we want to even avoid the object allocation of creating span objects, we can add a check in the Besu code that disables creating span objects.

Reply:
Object allocation is my worry, but I'm thinking if it matters it will only show up under load, which we have a test set up to detect (but only for committed code).

So if the test shows a regression the worst case scenario of a commit is a rollback and then a fix.

Reply:
I am willing to try this out and see how the performance is impacted.
Initially I had a span collecting trie performance, but I reverted it, as it was too verbose. I believe the AbstractBlockProcessor might be too verbose as well.
We'll see - it would be great to understand better if there are specific use cases (block imports, mining) where tracing helps specifically.